serialization/ruby/json/microsoft_kiota_serialization/lib/microsoft_kiota_serialization/json_parse_node.rb

```diff
@@ -0,0 +1,95 @@
+require 'microsoft_kiota_abstractions'
+require 'time'
+require 'json'
+require 'uuidtools'
+
+module MicrosoftKiotaSerialization
+  class JsonParseNode
+    include MicrosoftKiotaAbstractions::ParseNode
+    def initialize(node)
+      @current_node = node
+    end
+
+    def get_string_value
+      @current_node.to_s
+    end
+
+    def get_boolean_value
+      @current_node
+    end
+
+    def get_number_value
+      @current_node.to_i
+    end
+
+    def get_float_value
+      @current_node.to_f
+    end
+
+    def get_guid_value
+      UUIDTools::UUID.parse(@current_node)
+    end
+
+    def get_date_value
+      Time.parse(@current_node)
+    end
+
+    def get_collection_of_primitive_values(_type)
```

Comment:
I thought we said we'd switch on the type instead of raising exceptions here last time we chatted about this?